unity/Assets/Scripts/ActionDispatcher.cs

```diff
@@ -0,0 +1,233 @@
+
+using System.Reflection;
+using System.Collections.Generic;
+using System;
+using UnityEngine;
+
+
+/*
+    The ActionDispatcher takes a dynamic object with an 'action' property and 
+    maps this to a method.  Matching is performed using the parameter names. 
+    In the case of method overloading, the best match is returned based on the
+    number of matched named parameters.  For a method to qualify for dispatching
+    it must be public and have a return type of void.  The following method 
+    definitions are permitted:
+
+    public void MoveAhead()
+    public void MoveAhead(string direction)
+    public void MoveAhead(ServerAction action)
+    public void MoveAhead(float moveMagnitude, rotation=0.0f)
+
+
+    Creating the following overloaded set of functions will not work as expected:
+
+    public void Teleport(int x, int y)
+    public void Teleport(int x, short y)
+
+    as well the following scenario should also be avoided:
+
+    public void ObjectVisible(bool foo, int x, int y)
+    public void ObjectVisible(int x, int y, bool foo)
+
```

Comment:
What happens in this case? the sort function would just rank them the same way right? and choose one. maybe would be good to be able to distinguish this case from the correct override, by checking `==` case in the ranking function and propagate an error. What do you think?

Reply:
Yeah, this wouldn't fail. The getDispatchMethod would add both methods to the dispatchTable since the signatureMatch would be false between the two methods (we compare the parameters in order).  They would both end up with the same matchCount so we would just pick the first one.  I think for this case I don't want the error happening at runtime since that isn't real great for a user.  I am going to add a separate test that looks for these signature collisions and execute with the unit tests.  So if a collision is created, the build will fail and the committer will have to resolve the conflict?  Sound good?

Reply:
Sounds good!

Reply:
Okay, added a unit test to look for these types of conflicts.  If we need/want to have the aforementioned method overloading the dispatcher can be made more type aware and attempt to map the JSON types to csharp types to reduce the conflicts.